src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -32,46 +32,141 @@ namespace po = boost::program_options;
 
 // System - Include Files
 #include <iostream>
-
+#include <fstream>
 
 // =============================================================================
 
 // ------ L O C A L   F U N C T I O N S ---------------------------------------
 
 namespace {
 
-enum class configType {
+constexpr const char* config_file  = "/etc/msd.conf";
+
+enum class config_type {
     security = 0,
     clk_scaling,
     threshold_power_override,
+    reset,
 };
 
-enum class memType {
+enum class mem_type {
     unknown= 0,
     ddr,
     hbm, 
     
 };
 
+// Presumably a struct in anticipation of more data
+struct config
+{
+  std::string host;
+
+  void write(std::ostream& ostr)
+  {
+    ostr << "host=" << host << std::endl;
+  }
+};
+
+static std::string
+get_hostname()
+{
+  std::string hn;
+#ifdef __GNUC__
```

Comment:
We should get the hostname from `xrt_core::get_os_info(...)` or maybe we should consider adding device_queries?

Reply:
Yes, I wanted to consolidate this some place else, but xrt_core::get_os_info take a property tree, to me that is not the right API,

Reply:
I agree. I am moving away from xrt_core::get_xx_info(). I did that for electricals, thermals, etc. I'm directly using device queries for them in reports. Do you think we should create device_queries for system info as well?